tests/test_etoro.py

```diff
@@ -0,0 +1,224 @@
+        else:
+            return EToroMockServer.handle_delete(url)
+
+    @staticmethod
+    def handle_get(url: str):
+        # Parse the URL to determine which piece of canned data to return
+        if re.search(r"api\/v1\/instruments", url):
+            return MockedResponse(text=EToroMockServer.responses["markets"])
+        elif re.search(r"api\/v1\/balances", url):
+            return MockedResponse(text=EToroMockServer.responses["balances"])
+        elif re.search(r"api\/v1\/orders", url):
+            return MockedResponse(text=EToroMockServer.responses["orders"])
+        elif re.search(r"\/api\/v1\/trades", url):
+            return MockedResponse(text=EToroMockServer.responses["trades"])
+        else:
+            raise Exception("Unable to match HTTP GET request to canned response")
+
+    @staticmethod
+    def handle_post(url: str, data):
+        assert(data is not None)
+        if re.search(r"\/api\/v1\/orders", url):
+            return MockedResponse(text=EToroMockServer.responses["single_order"])
+        # elif re.search(r"\/api\/v1\/orders", url):
+        #     return MockedResponse(text=EToroMockServer.responses["place_order_failure"])
+        else:
+            raise Exception("Unable to match HTTP POST request to canned response")
+
+    
+    @staticmethod
+    def handle_delete(url: str):
+        if re.search(r"\/api\/v1\/orders\/[\w\-_]+", url):
+            return MockedResponse(text=EToroMockServer.responses["cancel_order"])
+        else:
+            raise Exception("Unable to match HTTP DELETE request to canned response")
+
+class TestEToro:
+    def setup_method(self):
+        cwd = os.path.dirname(os.path.realpath(__file__))
+        self.etoro = EToroApi(
+            api_server = "localhost",
+            api_key = "00000000-0000-0000-0000-000000000000",
+            secret_key = open(os.path.join(cwd, "mock/etoro-test-key"), "r").read(),
+            timeout = 15.5
+        )
+
+    def test_get_markets(self, mocker):
+        mocker.patch("requests.request", side_effect=EToroMockServer.handle_request)
+        response = self.etoro.get_markets()
+        assert(len(response) > 0)
+        assert(any(x["id"] == "ethusdc" for x in response))
+
+    def test_order(self):
+        price = Wad.from_number(4.8765)
+        amount = Wad.from_number(0.222)
+        remaining_amount = Wad.from_number(0.153)
+        order = Order(
+            order_id="153153",
+            timestamp=int(time.time()),
+            instrument_id="ethusdc",
+            is_sell=False,
+            price=price,
+            amount=amount,
+            remaining_amount=remaining_amount
+        )
+        assert(order.price == order.sell_to_buy_price)
+        assert(order.price == order.buy_to_sell_price)
+        # assert(order.remaining_buy_amount == amount-remaining_amount)
+        # assert(order.remaining_sell_amount == (amount-remaining_amount)*price)
```

Comment:
Perhaps you could replace these comments with `assert(order.remaining_amount == remaining_amount)`.

Reply:
Replaced with your recommendation.